counsel.el

```diff
@@ -2700,37 +2700,51 @@ And insert it into the minibuffer. Useful during
 (defvar counsel-linux-apps-faulty nil
   "List of faulty desktop files.")
 
+(defvar counsel--linux-apps-cache nil
+  "Cache of desktop files data.")
+
+(defvar counsel--linux-apps-cached-files nil
+  "List of cached desktop files.")
+
+(defvar counsel--linux-apps-cache-timestamp nil
+  "Time when we last updated the cached application list.")
+
 (defun counsel-linux-apps-list-desktop-files ()
```

Comment:
I changed this to return a stable list of desktop files so we can detect changes with a simple `(equals old-list new-list)`.